### PR TITLE
more ReactNode support, document children where helpful

### PR DESCRIPTION
**Pull request by giladgray**
MenuDivider, NIS, Tab, Controls

### CODE REVIEW

File: packages/core/src/components/menu/menuDivider.tsx [hunk 1]
```diff
@@ -11,23 +11,26 @@ import * as Classes from "../../common/classes";
 import { IProps } from "../../common/props";
 
 export interface IMenuDividerProps extends IProps {
-    /** Optional header title. */
-    title?: string;
+    /** Optional header title. Can also be passed as `children`. */
```
Comment (adidahiya) [inline packages/core/src/components/menu/menuDivider.tsx]: why both? it seems unnecessary

Comment (giladgray) [reply packages/core/src/components/menu/menuDivider.tsx]: it does indeed. but `children` was supported and unused. how would you like to see this resolved?

Comment (adidahiya) [reply packages/core/src/components/menu/menuDivider.tsx]: it was supported but not documented? then just remove it and say it's a breaking change

Comment (giladgray) [reply packages/core/src/components/menu/menuDivider.tsx]: the `children` prop was allowed, but was not rendered. now it is rendered.

Comment (adidahiya) [reply packages/core/src/components/menu/menuDivider.tsx]: cool, I'd rather remove the broken feature than add proper support for it

Comment (giladgray) [reply packages/core/src/components/menu/menuDivider.tsx]: wait... so... `children?: never` or remove `title` prop?

Comment (adidahiya) [reply packages/core/src/components/menu/menuDivider.tsx]: tbh I don't think typescript is able to fail on children passed to components with `children: never` props in `.tsx` files, so it's purely for documentation

**Pull request merged**